src/python/pants/goal/migrate_call_by_name.py

```diff
@@ -314,57 +214,149 @@ class RuleFunction(TypedDict):
-    col_range: tuple[int, int]
-    current_source: ast.Call
-    new_source: ast.Call
-    additional_imports: list[ast.ImportFrom]
-    # TODO: Use libcst or another CST, rather than an ast
-    add_trailing_comma: bool = False
-
-    def sanitized_imports(self) -> list[ast.ImportFrom]:
+    current_source: cst.Call
+    new_source: cst.Call
+    additional_imports: list[cst.ImportFrom]
+
+    def sanitized_imports(self) -> list[cst.ImportFrom]:
         """Remove any circular or self-imports."""
-        return [i for i in self.additional_imports if i.module != self.module]
+        cst_module = cstutil.make_importfrom_attr(self.module)
+        return [
+            i for i in self.additional_imports if i.module and not cst_module.deep_equals(i.module)
+        ]
 
-    def sanitize(self, names: set[str]):
+    def sanitize(self, unavailable_names: set[str]):
         """Remove any shadowing of names, except if the new_func is in the current file."""
-        assert isinstance(self.new_source.func, ast.Name)
-        func_name = self.new_source.func.id
-        if func_name not in names:
+
+        func_name = cst.ensure_type(self.new_source.func, cst.Name)
+        if func_name.value not in unavailable_names:
             return
 
-        # If the new function is not in the sanitized imports, it must be in the current file
-        if not any(i.names[0].name == func_name for i in self.sanitized_imports()):
+        # If the new func_name is not in the sanitized imports, it must already be in the current file
+        imported_names: set[str] = set()
+        for i in self.sanitized_imports():
+            assert isinstance(i.names, Iterable)
+            for import_alias in i.names:
+                alias_name = cst.ensure_type(import_alias.name, cst.Name)
+                imported_names.add(alias_name.value)
+
+        if func_name.value not in imported_names:
             return
 
-        bound_name = f"{func_name}_get"
-        self.new_source.func.id = bound_name
-        for i in self.additional_imports:
-            if i.names[0].name == func_name:
-                i.names[0].asname = bound_name
-        logging.warning(f"Renamed {func_name} to {bound_name} to avoid shadowing")
+        # In-place update this replacement and additional imports
+        bound_name = f"{func_name.value}_get"
+        self.new_source = self.new_source.with_deep_changes(self.new_source.func, value=bound_name)
+
+        for i, imp in enumerate(self.additional_imports):
+            assert isinstance(imp.names, Iterable)
+            for import_alias in imp.names:
+                alias_name = cst.ensure_type(import_alias.name, cst.Name)
+                if alias_name.value == func_name.value:
+                    self.additional_imports[i] = imp.with_changes(
+                        names=[cst.ImportAlias(func_name, asname=cst.AsName(cst.Name(bound_name)))]
+                    )
 
-    def contains_comments(self, source_code: str) -> bool:
```

Comment:
Not needed, handled by libcst. `ast` wipes out comments by default, `cst` keeps them